### PR TITLE
Initial release of private airship system

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: #3314

* **Server Mode**: Renewal

* **Description of Pull Request**: 
Adds support for the private airship system and it's packets.

### CODE REVIEW

File: src/map/clif.cpp [hunk 1]
```diff
@@ -20388,6 +20388,121 @@ void clif_weight_limit( struct map_session_data* sd ){
...
+	}
+
+	// Check if the player is allowed to warp to the target map
+	if( !map_getmapflag( mapid, MF_PRIVATEAIRSHIP_DESTINATION ) ){
+		clif_private_airship_response( sd, PRIVATEAIRSHIP_DESTINATION_MAP_INVALID );
+		return;
+	}
+
+	// The UI only offers these two buttons(items) for now
+	uint16 item_ids[] = {
```
Comment (Atemo) [inline src/map/clif.cpp]: The items ID are hardcoded in the client ?

Comment (Lemongrass3110) [reply src/map/clif.cpp]: Inside a lua file. The mapflags as well!

**Comment by Haikenz** [comment]
was missing item 25464 on pull

File: src/map/clif.cpp [hunk 2]
```diff
@@ -20388,6 +20388,121 @@ void clif_weight_limit( struct map_session_data* sd ){
...
+	int idx = pc_search_inventory( sd, item_id );
+
+	// Check if the player has the item at all
+	if( idx < 0 ){
+		clif_private_airship_response( sd, PRIVATEAIRSHIP_ITEM_NOT_ENOUGH );
+		return;
+	}
+
+	// Delete the chosen item
+	if( pc_delitem( sd, idx, 1, 0, 0, LOG_TYPE_PRIVATE_AIRSHIP ) ){
```
[the remark this answers is not in the document]
Comment (sctnightcore) [reply src/map/clif.cpp]: I get the same.

[the remark this answers is not in the document]
Comment (Haikenz) [reply src/map/clif.cpp]: me too

**Comment by ryuma1922** [comment]
I want information about updating map_getmapdata
Since I did not follow every update, it asked me to build

**Pull request merged**